tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java

```diff
@@ -69,19 +72,30 @@ public Controller(String muleHome, int timeout) {
     this.domainsDir = new File(muleHome + "/domains");
     this.appsDir = new File(muleHome + "/apps/");
     this.libsDir = new File(muleHome + "/lib/user");
+    this.internalRepository = new File(muleHome, "repository");
     this.wrapperConf = Paths.get(muleHome + "/conf/wrapper.conf");
     this.timeout = timeout != 0 ? timeout : DEFAULT_TIMEOUT;
   }
 
   public abstract String getMuleBin();
 
   public void start(String[] args) {
+    checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(args);
     int error = runSync("start", args);
     if (error != 0) {
       throw new MuleControllerException("The mule instance couldn't be started");
     }
   }
 
+  protected void checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(String... args) {
+    Optional<String> repoVar =
+        Arrays.asList(args).stream().filter(arg -> arg.contains("-M-DmuleRuntimeConfig.maven.repositoryLocation=")).findFirst();
+    if (repoVar.isPresent()) {
+      this.internalRepository = new File(repoVar.get().split("=")[1]);
```

Comment:
but as I said in the previous comment the confusion could come between the runtimeRepository and the artifactRepository please check this two methods:

   /**
    * @param artifactName
    * @return the directory of the internal repository for the artifact with the given name.
    */
   protected File getArtifactInternalRepository(String artifactName) {
     return new File(new File(appsDir, artifactName), "repository");
   }
 
   /**
     * @return the directory of the internal repository for the Mule runtime.
     */
    protected File getRuntimeInternalRepository() {
 -    return new File(muleHome, "repository");
 +    return this.internalRepository;